packages/core/src/new-api/internal/new-execution/future-processor/handlers/send-transaction.ts

```diff
@@ -77,7 +82,10 @@ export async function sendTransaction(
     lastNetworkInteraction,
     async (_sender: string) => nonceManager.getNextNonce(_sender),
     async (simulationResult) => {
-      const response = await strategyGenerator.next(simulationResult);
+      const response = await strategyGenerator.next({
+        type: OnchainInteractionResponseType.SIMULATION_RESULT,
+        result: simulationResult,
+      } as any);
```

Comment:
If this isn't wrapped up as a `SimulationResult`, the basic strategy errors on the first assertion, claiming it shouldn't recieve raw simulation results.

I was unsure what the intent was here, and so haven't changed the types.

How do you want me to approach this?

Reply:
Constraining the generator resolved the issue.